src/model/ChillerElectricASHRAE205.hpp

```diff
@@ -0,0 +1,231 @@
+    void resetAmbientTemperatureOutdoorAirNodeName();
+
+    bool setChilledWaterMaximumRequestedFlowRate(double chilledWaterMaximumRequestedFlowRate);
+    void autosizeChilledWaterMaximumRequestedFlowRate();
+
+    bool setCondenserMaximumRequestedFlowRate(double condenserMaximumRequestedFlowRate);
+    void autosizeCondenserMaximumRequestedFlowRate();
+
+    bool setChillerFlowMode(const std::string& chillerFlowMode);
+
+    bool setOilCoolerDesignFlowRate(double oilCoolerDesignFlowRate);
+    void resetOilCoolerDesignFlowRate();
+
+    bool setAuxiliaryCoolingDesignFlowRate(double auxiliaryCoolingDesignFlowRate);
+    void resetAuxiliaryCoolingDesignFlowRate();
+
+    bool setEndUseSubcategory(const std::string& endUseSubcategory);
+    void resetEndUseSubcategory();
+
+    //@}
+    /** @name Other */
+    //@{
+
+    boost::optional<double> autosizedRatedCapacity();
+
+    boost::optional<double> autosizedChilledWaterMaximumRequestedFlowRate();
+
+    boost::optional<double> autosizedCondenserMaximumRequestedFlowRate();
+
+    /** Convenience Function to return the Chilled Water Loop (chiller on supply) **/
+    boost::optional<PlantLoop> chilledWaterLoop() const;
+    // Same as supplyInletModelObject, but cast to a Node
+    boost::optional<Node> chilledWaterInletNode() const;
+    boost::optional<Node> chilledWaterOutletNode() const;
+
+    /** Convenience Function to return the Condenser Water Loop (chiller on demand side) **/
+    boost::optional<PlantLoop> condenserWaterLoop() const;
+    boost::optional<Node> condenserInletNode() const;
+    boost::optional<Node> condenserOutletNode() const;
+
+    // Not implemented in E+ 22.2.0, but reserved as a Tertiary Loop for now
+    boost::optional<PlantLoop> heatRecoveryLoop() const;
+    boost::optional<Node> heatRecoveryInletNode() const;
+    boost::optional<Node> heatRecoveryOutletNode() const;
+
+    unsigned oilCoolerInletPort() const;
+    boost::optional<ModelObject> oilCoolerInletModelObject() const;
+    boost::optional<Node> oilCoolerInletNode() const;
+    unsigned oilCoolerOutletPort() const;
+    boost::optional<ModelObject> oilCoolerOutletModelObject() const;
+    boost::optional<Node> oilCoolerOutletNode() const;
+    boost::optional<PlantLoop> oilCoolerLoop() const;
+    bool addDemandBranchOnOilCoolerLoop(PlantLoop& plantLoop);
+    bool addToOilCoolerLoopNode(Node& node);
+    bool removeFromOilCoolerLoop();
+
+    unsigned auxiliaryInletPort() const;
+    boost::optional<ModelObject> auxiliaryInletModelObject() const;
+    boost::optional<Node> auxiliaryInletNode() const;
+    unsigned auxiliaryOutletPort() const;
+    boost::optional<ModelObject> auxiliaryOutletModelObject() const;
+    boost::optional<Node> auxiliaryOutletNode() const;
+    bool addDemandBranchOnAuxiliaryLoop(PlantLoop& plantLoop);
+    bool addToAuxiliaryLoopNode(Node& node);
+    bool removeFromAuxiliaryLoop();
+    boost::optional<PlantLoop> auxiliaryLoop() const;
```

Comment:
Convenience methods, since this is very complicated. Also, `addDemandBranchOnOilCoolerLoop`, `addToOilCoolerLoopNode`, and  `removeFromOilCoolerLoop` (same for Auxiliary Loop) is provided to handle connections to loops